### PR TITLE
Ensure all assets are compiled on test page visit

### LINKED ISSUE #11010: [Flaky] rspec ./spec/system/consumer/caching/shops_caching_spec.rb:17
### What we should change and why (this is tech debt)

`rspec ./spec/system/consumer/caching/shops_caching_spec.rb:17`

```
Failures:

  1) Shops caching caching enterprises AMS data caches data for all enterprises, with the provided options
     Failure/Error: expect(Rails.cache.exist?(key, options)).to be true
     
       expected true
            got false
     
     [Screenshot Image]: /home/runner/work/openfoodnetwork/openfoodnetwork/tmp/capybara/screenshots/failures_r_spec_example_groups_shops_caching_caching_enterprises_ams_data_caches_data_for_all_enterprises__with_the_provided_options_155.png
```

### Context

https://github.com/openfoodfoundation/openfoodnetwork/actions/runs/5263043188/jobs/9512817149
https://openfoodnetwork.slack.com/archives/C012LE8LLDS/p1686718490274599?thread_ts=1686601349.929099&cid=C012LE8LLDS

### PR DESCRIPTION
**Pull request by mkllnk**
#### What? Why?

- Closes #11010 
- Replaces #12552

For best test performance, I chose to trigger on-demand asset compilation in a `before` block. Always pre-compiling all assets was wasting time because they are often not all used.

#### What should we test?

- Specs only.

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: spec/system/consumer/caching/shops_caching_spec.rb [hunk 1]
```diff
@@ -14,12 +14,13 @@
   }
 
   describe "caching enterprises AMS data" do
-    it "caches data for all enterprises, with the provided options" do
+    before do
       # Trigger lengthy tasks like JS compilation before testing caching:
       visit shops_path
       Rails.cache.clear
```
Comment (dacook) [inline spec/system/consumer/caching/shops_caching_spec.rb:20]: 🤔 I wonder if we can't trigger JS compilation directly?

**Review by dacook — APPROVED** [review]
Yay, another flaky spec down!

File: spec/system/consumer/caching/shops_caching_spec.rb (hunk 1 above)
Comment (mkllnk) [reply spec/system/consumer/caching/shops_caching_spec.rb:20]: Well, my previous PR was triggering the whole JS compilation but that was maybe inefficient. Visiting a certain page ensures that we compile only the assets needed for that page.

By the way, I'm not sure if my assessment of significant runtime increase on my previous PR was right. I've seen plenty of other builds taking that long as well. I may have just been unlucky with the build. I didn't look at the detailed spec run times.

**Review by rioug — APPROVED** [review]
Nice!

**Pull request merged**